test/blackbox-tests/test-cases/deprecated-library-name/features.t

```diff
@@ -96,8 +96,7 @@ that wasn't found:
   1 | (executable (name prog) (libraries a))
                                          ^
   Error: Library "a" not found.
-  -> required by _build/default/c/.prog.eobjs/byte/dune__exe__Prog.cmi
-  -> required by _build/default/c/.prog.eobjs/native/dune__exe__Prog.cmx
+  -> required by _build/default/c/.merlin-conf/exe-prog
```

Comment:
My main concern is that this changes so many tests, but given the dependency that is getting reported is an internal one and stays an internal one I believe this is not an issue.

Reply:
It looks like the PR changes some evaluation order that makes Dune errors after following a different branch. Not sure what causes this however...

Reply:
@rgrinberg Can you please weight in whether this is an issue? It doesn't seem to be wrong, but maybe the order is important in some way I am not aware of?

Reply:
I found the difference: before we only "peeked" (with [`Resolve.peek`](https://github.com/ocaml/dune/pull/10535/files#diff-d738a29d0c19436c2bb3dbf5358d8528183ba8558ed380f1bd6167a915e52c06L440-L444)) at the dependencies and now we ["read"](https://github.com/ocaml/dune/pull/10535/files#diff-d738a29d0c19436c2bb3dbf5358d8528183ba8558ed380f1bd6167a915e52c06R665-R666) them.

I will revert that change as is seems wrong to introduce dependencies here.